beprof/profile.py

```diff
@@ -132,23 +132,36 @@ def fwhm(self):
         """
         return self.width(0.5 * np.max(self.y))
 
-    def normalize(self, dt):
+    def normalize(self, dt, allow_cast=True):
         """
-        Normalize to 1 over [-dt, +dt] area
+        Normalize to 1 over [-dt, +dt] area, if allow_cast is set
+        to True, division not in place and casting may occur.
+        If division in place is not possible and allow_cast is False
+        an exception is raised.
         :param dt:
-        :return:
+        :param allow_cast:
         """
+        if dt <= 0:
+            raise ValueError("Expected positive input")
         logger.info('Running {0}.normalize(dt={1})'.format(self.__class__, dt))
         try:
             ave = np.average(self.y[np.fabs(self.x) <= dt])
         except RuntimeWarning as e:
             logger.error('in normalize(). self class is {0}, dt={1}'.format(self.__class__, dt))
-            raise Exception("Scaling factor error:\n" + str(e))
-        self.y /= ave
+            raise Exception("Scaling factor error: {0}".format(e))
+        try:
+            self.y /= ave
+        except TypeError as e:
+            logger.warning("Division in place is impossible: {0}".format(e))
+            if allow_cast:
+                self.y = self.y / ave
+            else:
+                logger.error("Division in place impossible - allow_cast flag set to True should help")
+                raise e
```

Comment:
Is it the correct way of re-throwing an exception ?
See i.e. http://blog.bstpierre.org/python-exception-handling-cleanup-and-reraise